FirebaseStorageSwift/Tests/Integration/StorageAsyncAwait.swift

```diff
@@ -0,0 +1,391 @@
+      XCTAssertEqual(meta.customMetadata!["lol"], metadata.customMetadata!["lol"])
+      XCTAssertEqual(meta.customMetadata!["ちかてつ"], metadata.customMetadata!["ちかてつ"])
+      XCTAssertEqual(meta.customMetadata!["shinkansen"],
+                     metadata.customMetadata!["shinkansen"])
+    }
+
+    func testDelete() async throws {
+      let ref = storage.reference(withPath: "ios/public/fileToDelete")
+      let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+      let result = try await ref.putDataAsync(data)
+      XCTAssertNotNil(result)
+      _ = try await ref.delete()
+    }
+
+    func testDeleteWithNilCompletion() async throws {
+      let ref = storage.reference(withPath: "ios/public/fileToDelete")
+      let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+      let result = try await ref.putDataAsync(data)
+      XCTAssertNotNil(result)
+    }
+
+    func testSimplePutData() async throws {
+      let ref = storage.reference(withPath: "ios/public/testBytesUpload")
+      let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+      let result = try await ref.putDataAsync(data)
+      XCTAssertNotNil(result)
+    }
+
+    func testSimplePutSpecialCharacter() async throws {
+      let ref = storage.reference(withPath: "ios/public/-._~!$'()*,=:@&+;")
+      let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+      let result = try await ref.putDataAsync(data)
+      XCTAssertNotNil(result)
+    }
+
+    func testSimplePutDataInBackgroundQueue() async throws {
+      actor MyBackground {
+        func doit(_ ref: StorageReference) async throws -> StorageMetadata {
+          let data = try XCTUnwrap(
+            "Hello Swift World".data(using: .utf8),
+            "Data construction failed"
+          )
+          XCTAssertFalse(Thread.isMainThread)
+          return try await ref.putDataAsync(data)
+        }
+      }
+      let ref = storage.reference(withPath: "ios/public/testBytesUpload")
+      let result = try await MyBackground().doit(ref)
+      XCTAssertNotNil(result)
+    }
+
+    func testSimplePutEmptyData() async throws {
+      let ref = storage.reference(withPath: "ios/public/testSimplePutEmptyData")
+      let data = Data()
+      let result = try await ref.putDataAsync(data)
+      XCTAssertNotNil(result)
+    }
+
+    func testSimplePutDataUnauthorized() async throws {
+      let ref = storage.reference(withPath: "ios/private/secretfile.txt")
+      let data = try XCTUnwrap("Hello Swift World".data(using: .utf8), "Data construction failed")
+      do {
+        _ = try await ref.putDataAsync(data)
+        XCTFail("Unexpected success from unauthorized putData")
+      } catch {
+        XCTAssertEqual((error as NSError).code, StorageErrorCode.unauthorized.rawValue)
+      }
+    }
+
+    func testSimplePutFile() throws {
```

Comment:
Should this test be removed? It doesnt look like it tests any async/await APIs

Reply:
maybe leave empty with the comment that this is for Task related API testing

Reply:
Yep!